allennlp/models/encoder_decoders/wikitables_semantic_parser.py

```diff
@@ -1127,13 +1143,38 @@ def take_step(self,
             # entity_action_mask: (group_size, num_entity_actions)
             entity_action_logits, entity_action_mask, entity_type_embeddings = \
                     self._get_entity_action_logits(state, actions_to_link, attention_weights)
-            action_logits = torch.cat([embedded_action_logits, entity_action_logits], dim=1)
-            action_mask = torch.cat([embedded_action_mask, entity_action_mask], dim=1).float()
 
             # The `action_embeddings` tensor gets used later as the input to the next decoder step.
             # For linked actions, we don't have any action embedding, so we use the entity type
             # instead.
             action_embeddings = torch.cat([action_embeddings, entity_type_embeddings], dim=1)
+
+            if self._mixture_feedforward is not None:
+                # The entity and action logits are combined with a mixture weight to prevent the
+                # entity_action_logits from dominating the embedded_action_logits if a softmax
+                # was applied on both together.
+                # Todo(rajas) remove mix1 and mix2 variables
+                # Todo(rajas) figure out why the loss is nan with this code.
+                mixture_weight = self._mixture_feedforward(hidden_state)
+                mix1 = (torch.log(mixture_weight) * entity_action_mask.float())
+                mix2 = (torch.log(1 - mixture_weight) * embedded_action_mask.float())
+                entity_action_logits = util.masked_log_softmax(entity_action_logits, entity_action_mask.float()) + mix1
```

Comment:
This looks right, I'm not sure why you would be getting nans here.  Maybe some value is just initialized too high, and so training is unstable?  You could try printing out the mixture weight and the values of mix1 and mix2, to see if something looks wrong.

Also, because you've done a log softmax, these are no longer logits, they are log probabilities.